org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configurationevent/AbstractConfigurationEventAssert.java

```diff
@@ -0,0 +1,235 @@
+	 *            ConfigurationEvent's factoryPid to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's factoryPid is
+	 *             not equal to the given one.
+	 */
+	public S hasFactoryPid(String factoryPid) {
+
+		isNotNull().has(ConfigurationEventConditions.factoryPid(factoryPid));
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's factoryPid is not null.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's factoryPid is
+	 *             null.
+	 */
+	public S hasFactoryPid() {
+
+		isNotNull().has(ConfigurationEventConditions.factoryPidNotNull());
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's factoryPid is null.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's factoryPid is
+	 *             not null.
+	 */
+	public S hasNoFactoryPid() {
+
+		isNotNull().has(ConfigurationEventConditions.factoryPidNull());
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's pid is not null.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's pid is null
+	 */
+	public S hasPid() {
+
+		isNotNull().has(ConfigurationEventConditions.pidNotNull());
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's pid is null.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's pid is not
+	 *             null.
+	 */
+	public S hasNoPid() {
+
+		isNotNull().has(ConfigurationEventConditions.pidNull());
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's pid is equal to the given
+	 * one.
+	 *
+	 * @param pid the given pid to compare the actual ConfigurationEvent's pid
+	 *            to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's pid is not
+	 *             equal to the given one.
+	 */
+	public S hasPid(String pid) {
+
+		isNotNull().has(ConfigurationEventConditions.pid(pid));
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's reference is equal to the
+	 * given one.
+	 *
+	 * @param reference the given reference to compare the actual
+	 *            ConfigurationEvent's reference to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's reference is
+	 *             not equal to the given one.
+	 */
+	public S hasReference(org.osgi.framework.ServiceReference<ConfigurationAdmin> reference) {
```

Comment:
We don't need this since you have `hasReferenceThat`. And besides the equality test of deepEquals seems wrong.